services/121-service/src/transaction-jobs/services/transaction-jobs-safaricom.service.spec.ts

```diff
@@ -52,31 +43,36 @@ describe('TransactionJobsSafaricomService', () => {
         SafaricomTransferScopedRepository,
       );
 
-    transactionScopedRepository = unitRef.get<TransactionScopedRepository>(
-      TransactionScopedRepository,
-    );
+    transactionEventsScopedRepository =
+      unitRef.get<TransactionEventsScopedRepository>(
+        TransactionEventsScopedRepository,
+      );
     transactionJobsHelperService = unitRef.get<TransactionJobsHelperService>(
       TransactionJobsHelperService,
     );
+    transactionsService = unitRef.get<TransactionsService>(TransactionsService);
 
-    jest
-      .spyOn(transactionJobsHelperService, 'getRegistrationOrThrow')
-      .mockResolvedValue(mockedRegistration);
     jest
       .spyOn(
         transactionJobsHelperService,
-        'createTransactionAndUpdateRegistration',
+        'createInitiatedOrRetryTransactionEvent',
       )
-      .mockResolvedValue({ id: mockedTransactionId } as any);
+      .mockImplementation();
     jest
       .spyOn(safaricomTransferScopedRepository, 'findOne')
       .mockResolvedValue(undefined);
+    jest
+      .spyOn(safaricomTransferScopedRepository, 'update')
+      .mockResolvedValue({} as UpdateResult);
     jest
       .spyOn(safaricomTransferScopedRepository, 'save')
       .mockImplementation(async (entity) => ({ ...entity, id: 1 }));
     jest
-      .spyOn(transactionScopedRepository, 'update')
-      .mockResolvedValue({} as UpdateResult);
+      .spyOn(
+        transactionEventsScopedRepository,
+        'countFailedTransactionAttempts',
+      )
+      .mockResolvedValue(0);
```

Comment:
I was reading the unit test in this file and I don't get it. It's called: 
`Idempotency] safaricom transaction job processing should fail when using same originatorConversationId` 

But it does not test if that error is thrown or if you get a fail or something. 

Not sure if we should improve this test or remove it entirely as the created message is also tested in `services/121-service/src/payments/fsp-integration/safaricom/services/safaricom.api.helper.service.spec.ts`